### ISSUE TITLE
Long treePath value can block import

### ISSUE THREAD

**Issue by ralfpuchert** [camunda/camunda#32891]
### Describe the bug
If the value in field `treePath` in index `list-view` gets to big (>32K) it blocks the import.

###  To Reproduce

- [ ] Create a tree path with value of >32K characters.
- [ ] Try to write the value in `list-view` index.

### Current behavior
* Import is blocked. It retries forever to import the data but fails due to exception during the writing of the field `treePath`.

### Expected behavior
* Check the expected tree path text length. 
* Use only the first part (process instance id) for the value.
* Log warn message about truncated tree path with affected process instance.

### Environment
SaaS

### Support cases:
* https://jira.camunda.com/browse/SUPPORT-17394
* https://jira.camunda.com/browse/SUPPORT-27123
---------------------------------------------------------------------------------------------

### Definition of Ready - Checklist

- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] The issue has been labeled with an appropriate `Bug-area` label
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

### :point_right: Handover Dev to QA 

- Changed components:
- Side effects on other components:

- Handy resources: 
     BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc : 
     
- Example projects:

- Commands needed to test; Versions to validate:

- Docker file : in case that it needed to be tested via docker share the version contain the fixed along with version of other services . 

- Release version ( in which version this fixed/feature will be released):

**Comment by RomanJRW** [comment]
Assigning to @yohanfernando 

@yohanfernando - note that some more discovery here needs to be done when considering the solution approach (see thread [here](https://camunda.slack.com/archives/C08F5RD5X8B/p1749213515463679?thread_ts=1749204318.247899&cid=C08F5RD5X8B)).

Feel free to ping me when you get working on this and we can kick it off together

**Comment by RomanJRW** [comment]
Pinged @theburi to get input on product impact of proposed change

**Comment by RomanJRW** [comment]
This is a long standing issue, not unique to 8.8. We do not consider it a release blocker for the 8.8 release

**Comment by EuroLew** [comment]
Analysis:

There is a `treePath` field in the list-view index which uses the `ca_path_tree` analyzer which uses a `ca_hierarchy` tokenizer with `path_hierarchy` with "/" as a delimeter. 

THis works by breaking down a path into all possible parent paths e.g. for `/root/a/b/doc` you woudl generate these tokens:
- `/root`
- `/root/a`
- `/root/a/b`
- `/root/a/b/doc`
this is useful for hierachical filtering, counting documents at different levels combined with joinRelation this greatly enriches querying.

The error that occurs is when this field is too long, ES has a limit of 32766 bytes and if the tokenizer generates at least one token exceeding this limit it throws an error and blocks the importer due to the nature of importer retry logic (or at least slows down the importer a LOT).

Solutions:
- `ignore_above` on `treePath` does not work as its only for `keyword` fields and it is currently `text`.
- the length approach where a length filter is added
```
        "analyzer": {
          "ca_path_tree": {
            "tokenizer": "ca_hierarchy",
            "filter": ["length_30000"]
          }

"filter": {
  "length_30000": {
    "type": "length",
    "min": 0,
    "max": 30000
  }
}
```
This runs AFTER tokenization and discards tokens that exceed 30k chars, it's simply removed from the index, this prevents errors while keeping shorter paths searchable

- another solution is to use `max_token_length` parameter for the tokenizer

```
"tokenizer": {
  "ca_hierarchy": {
    "type": "path_hierarchy",
    "delimiter": "/",
    "max_token_length": 30000
  }
}
```
But this has a drawback of trncating the token at that specific length and the truncated portion is lost, this can created invalid paths like `root/a/b/abc...`

**Comment by RomanJRW** [comment]
Thanks for the analysis @EuroLew 👍 

In terms of impact in the Operate UI (navigating between processes), do you have an understanding of how users would be impacted by each solution? I think this is likely to variable that will determine how we approach solving this

**Comment by theburi** [comment]
@aleksander-dytko Can you consider the customer impact of this issue?

**Comment by eppdot** [comment]
@mustafadagher @RomanJRW is this already solved/done? should this be considered relevant for the 8.9 release?

_(sidenote: I am updating the [quality board](https://github.com/orgs/camunda/projects/187/views/20) for 8.9 delivery)_

**Comment by RomanJRW** [comment]
Yes but I think it was tracked in this (https://github.com/camunda/camunda/issues/35648). Will make it as closed, as we do not intend to backport it
